### PR TITLE
Location Reassignment: Improvements to checks for site codes

### PR DESCRIPTION
**Pull request by mkangia**
##### SUMMARY
Some improvements for site codes
1. [Ticket](https://dimagi-dev.atlassian.net/browse/QA-1223?focusedCommentId=48380) -> always consider them as strings. A site code is meant to be a string and using them as integers, if received as integers from excel workbook can lead to errors
2. They should always be lower case
3. [Ticket](https://dimagi-dev.atlassian.net/browse/QA-1265) -> they should not have special characters

##### FEATURE FLAG
`LOCATION_REASSIGNMENT`

### CODE REVIEW

**Comment by mkangia** [comment]
@orangejenny can you take a look at this today sometime? this should be quick.

File: corehq/apps/locations/util.py [hunk 1]
```diff
@@ -336,3 +337,8 @@ def get_locations_from_ids(location_ids, domain, base_queryset=None):
     if len(locations) != expected_count:
         raise SQLLocation.DoesNotExist('One or more of the locations was not found.')
     return locations
+
+
+def valid_location_site_code(site_code):
```
Comment (orangejenny) [inline corehq/apps/locations/util.py:343]: Non-blocking: I'd expected this to be part of `parse_site_code` and throw an exception that you could catch in `Parser.parse`.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/locations/util.py (hunk 1 above)
Comment (mkangia) [reply corehq/apps/locations/util.py:343]: I tried seeing how I could accommodate and realised about how i have structured the parser that makes its a bit difficult.
In parser, everything necessary is read first in form of `TransitionRow`. Site codes are saved on it, and validated later. It's like a 2 step.
Hence the first step just does the parsing and ensures the site code is lower case which is not related to validation. It's replicating what is currently done in the locations view. I was originally thinking of bringing that behaviour in `validate_location_site_code` itself but i wanted to refrain from making changes to the core HQ code.

Another way would be to delay even the parsing to when the validation is finally called, but then i would handle that transformation(lowercase) when initialising the `Transition` object, which does not look better than already keeping parsed site codes on `TransitionRow`.

Is it that any `parse` method should also `validate` which is causing the confusion here? Or should that method be named something else other than `parse_site_code`, may be `sanitize_site_code`?

Comment (mkangia) [reply corehq/apps/locations/util.py:343]: Going to merge the branch so that this gets in next deploy but happy to take as a follow-up PR

Comment (orangejenny) [reply corehq/apps/locations/util.py:343]: I think the line between parsing and validation is blurry, and reasonable developers can disagree on where to put the alphanumeric check. No need to rename, this is fine as it is.

**Pull request merged**